hbase-server/src/main/java/org/apache/hadoop/hbase/io/FileLink.java

```diff
@@ -312,14 +314,31 @@ private FSDataInputStream tryOpen() throws IOException {
           }
           currentPath = path;
           return(in);
-        } catch (FileNotFoundException e) {
-          // Try another file location
-        } catch (RemoteException re) {
-          IOException ioe = re.unwrapRemoteException(FileNotFoundException.class);
-          if (!(ioe instanceof FileNotFoundException)) throw re;
+        } catch (FileNotFoundException | AccessControlException | RemoteException e) {
+          exception = handleTryOpenException(e, exception);
+        }
+      }
+      throw exception;
+    }
+
+    private IOException handleTryOpenException(IOException newException,
```

Comment:
Why need to save the previousException ?  Can just check the given exception e, and if  e or its cause is FileNotFoundException or AccessControlException, then return,  else just throw ?

Reply:
Because if fails to open all files, throw AccessControlException if it occurs when open any one of the files, otherwise, throw FileNotFoundException.